src/containerapp/azext_containerapp/containerapp_env_telemetry_decorator.py

```diff
@@ -151,3 +152,172 @@ def update(self):
+    def get_argument_enable_open_telemetry_metrics(self):
+        return self.get_param("enable_open_telemetry_metrics")
+
+    def construct_payload(self):
+        # Get current containerapp env telemetry properties
+        try:
+            self.existing_managed_env_def = self.client.show(cmd=self.cmd, resource_group_name=self.get_argument_resource_group_name(), name=self.get_argument_name())
+        except Exception as e:
+            handle_raw_exception(e)
+
+        self.set_up_otlp()
+        self.set_up_enable_open_telemetry_traces()
+        self.set_up_enable_open_telemetry_logs()
+        self.set_up_enable_open_telemetry_metrics()
+
+    def construct_remove_payload(self):
+        # Get current containerapp env telemetry properties
+        try:
+            self.existing_managed_env_def = self.client.show(cmd=self.cmd, resource_group_name=self.get_argument_resource_group_name(), name=self.get_argument_name())
+        except Exception as e:
+            handle_raw_exception(e)
+
+        otlp_name = self.get_argument_otlp_name()
+        existing_otlps = safe_get(self.existing_managed_env_def, "properties", "openTelemetryConfiguration", "destinationsConfiguration", "otlpConfigurations")
+        if existing_otlps is not None:
+            otlp = [p for p in existing_otlps if p["name"].lower() == otlp_name.lower()]
+            if otlp is not None:
+                idx = [i for i, p in enumerate(existing_otlps) if p["name"].lower() == otlp_name.lower()][0]
+                otlp_to_remove = existing_otlps[idx]
+                existing_otlps.remove(otlp_to_remove)
+                safe_set(self.managed_env_def, "properties", "openTelemetryConfiguration", "destinationsConfiguration", "otlpConfigurations", value=existing_otlps)
+            else:
+                raise ValidationError(f"No otlp entry with --otlp-name {otlp_name} found")
+        else:
+            raise ValidationError(f"No otlp entry with --otlp-name {otlp_name} found")
+        
+        self.set_up_enable_open_telemetry_traces()
+        self.set_up_enable_open_telemetry_logs()
+        self.set_up_enable_open_telemetry_metrics()
+        
+    def set_up_otlp(self):
+        if self.get_argument_otlp_name() is not None:
+
+            # Get current containerapp env telemetry properties
+            try:
+                self.existing_managed_env_def = self.client.show(cmd=self.cmd, resource_group_name=self.get_argument_resource_group_name(), name=self.get_argument_name())
+            except Exception as e:
+                handle_raw_exception(e)
+
+            otlp_name = self.get_argument_otlp_name()
+            endpoint = self.get_argument_endpoint()
+            insecure = self.get_argument_insecure()
+            headers = self.get_argument_headers()
+            existing_otlps = safe_get(self.existing_managed_env_def, "properties", "openTelemetryConfiguration", "destinationsConfiguration", "otlpConfigurations")
+            otlp = {"name": otlp_name} 
```

Comment:
>otlp_name [](http://example.com/codeflow?start=28&length=9)

is it possible otlp_name is ""

Reply:
otlp_name is required params so it should be ""